src/components/Tool/FixPhoneNumbers/FixPhoneNumbers.tsx

```diff
@@ -83,98 +90,78 @@ const useStyles = makeStyles()(() => ({
 
 export interface ModalState {
   open: boolean;
-  personId: string;
+  personIndex: number;
   numberIndex: number;
   phoneNumber: string;
 }
 
 const defaultDeleteModalState = {
   open: false,
-  personId: '',
+  personIndex: 0,
   numberIndex: 0,
   phoneNumber: '',
 };
 
-export interface PhoneNumberData {
-  id?: string;
-  primary: boolean;
-  updatedAt: string;
-  source: string;
-  number: string;
-  destroy?: boolean;
-}
-
 export interface PersonPhoneNumbers {
-  phoneNumbers: PhoneNumberData[];
-  toDelete: PersonPhoneNumberInput[];
+  phoneNumbers: PersonPhoneNumberFragment[];
 }
 
 interface Props {
   accountListId: string;
   setContactFocus: SetContactFocus;
 }
 
+interface FormValuesPerson extends PersonInvalidNumberFragment {
+  newPhoneNumber: PersonPhoneNumberFragment | Record<string, never> | string;
+  isNewPhoneNumber: boolean;
+}
+
+export interface FormValues {
+  people: FormValuesPerson[] | [];
+}
+
 const FixPhoneNumbers: React.FC<Props> = ({
   accountListId,
   setContactFocus,
 }: Props) => {
   const { classes } = useStyles();
   const { enqueueSnackbar } = useSnackbar();
-  const [defaultSource, setDefaultSource] = useState('MPDX');
+  const { appName } = useGetAppSettings();
+  const [defaultSource, setDefaultSource] = useState<string | undefined>(
+    appName || 'MPDX',
+  );
   const [deleteModalState, setDeleteModalState] = useState<ModalState>(
     defaultDeleteModalState,
   );
   const [updateInvalidPhoneNumbers] = useUpdateInvalidPhoneNumbersMutation();
-
   const { data, loading } = useGetInvalidPhoneNumbersQuery({
     variables: { accountListId },
   });
-  const [dataState, setDataState] = useState<{
-    [key: string]: PersonPhoneNumbers;
-  }>({});
   const { t } = useTranslation();
 
-  useEffect(
-    () =>
-      setDataState(
-        data
-          ? data.people.nodes?.reduce<{ [key: string]: PersonPhoneNumbers }>(
-              (map, person) => ({
-                ...map,
-                [person.id]: {
-                  phoneNumbers: person.phoneNumbers.nodes.map(
-                    (phoneNumber) => ({
-                      id: phoneNumber.id,
-                      primary: phoneNumber.primary,
-                      updatedAt: phoneNumber.updatedAt,
-                      source: phoneNumber.source,
-                      number: phoneNumber.number || '',
-                    }),
-                  ),
-                  toDelete: [],
-                },
-              }),
-              {},
-            )
-          : {},
-      ),
-    [loading],
-  );
+  const initialValues: FormValues = {
+    people:
+      data?.people?.nodes.map((person) => ({
+        ...person,
+        isNewPhoneNumber: false,
+        newPhoneNumber: '',
+      })) || [],
+  };
 
   const determineBulkDataToSend = (
```

Comment:
Does this helper depend on anything inside of the component? If not, can we move it outside of the component? Or even into a separate helpers file?

Reply:
Another benefit is that it makes it a lot easier to test logic when it's in a helper method instead of inside a component. Can you add some tests after you extract the logic into a separate file?

Reply:
If your function is in `helpers.ts`, the tests for it should go in `helpers.test.ts`.